app/models/course/reference_timeline.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+class Course::ReferenceTimeline < ApplicationRecord
+  belongs_to :course, inverse_of: :reference_timelines
+  has_many :reference_times,
+           class_name: Course::ReferenceTime.name, inverse_of: :reference_timeline, dependent: :destroy
+
+  validates :default, inclusion: [true, false]
+
+  def initialize_duplicate(duplicator, _other)
+    self.course = duplicator.options[:destination_course]
+    self.reference_times = []
```

Comment:
Not sure if this is correct, if I understand correctly this relies on the duplication of objects occurring in the following order:

1. ReferenceTimeline
2. LessonPlanItem
3. ReferenceTime

Since ReferenceTimeline and LessonPlanItem are both top-level course items, I think there may be some cases where LessonPlanItem may be duplicated before ReferenceTimeline

https://github.com/coursemology-collab/coursemology2/blob/master/lib/autoload/duplicator.rb#L26-L35

Reply:
The order is fixed here: https://github.com/Coursemology/coursemology2/pull/3130/files#diff-fee51348e72b919d2b3ab33af738299cR18

Reference timeline gets duplicated first

Reply:
Actually, should `reference_timeline` and `reference_time` even be duplicated? `reference_timelines` will be tied to course users, which are different across courses.

Reply:
Reference timeline is not the personalized timeline for each user. For most cases there will be exactly one reference timeline and we need to duplicate this else we lose the deadline info.

Reply:
Reference timeline is not the personalized timeline for each user. For most cases there will be exactly one reference timeline and we need to duplicate this else we lose the deadline info.